middleware/proxy/upstream.go

```diff
@@ -147,6 +153,15 @@ func parseBlock(c *parse.Dispenser, u *staticUpstream) error {
 			return err
 		}
 		u.MaxFails = int32(n)
+	case "max_conns":
+		if !c.NextArg() {
+			return c.ArgErr()
+		}
+		n, err := strconv.Atoi(c.Val())
+		if err != nil {
+			return err
+		}
+		u.MaxConns = int64(n)
```

Comment:
since `MaxConns` is `int64`. I'd go with `strconv.ParseInt(c.Val(), 10, 64)` instead.